// Copyright 2019 The GraphicsFuzz Project Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package gfauto;

import "gfauto/common.proto";


// An artifact is a directory containing an ArtifactMetadata proto stored in "artifact.json" and/or a Recipe proto
// stored in "recipe.json", plus any other files and directories within. The recipe is executed to produce the files
// that make up the artifact, which must include the "artifact.json" file.
// See artifact.proto.
message Recipe {

  oneof recipe {
    RecipeDownloadAndExtractArchiveSet download_and_extract_archive_set = 1;
  }
}

// Downloads and extracts one or more archives and makes the binaries available.
message RecipeDownloadAndExtractArchiveSet {
  ArchiveSet archive_set = 1;
}
